module/apmbuffalo/example/actions/app.go

```diff
@@ -0,0 +1,80 @@
+package actions
+
+import (
+	"github.com/gobuffalo/buffalo"
+	"github.com/gobuffalo/envy"
+	forcessl "github.com/gobuffalo/mw-forcessl"
+	paramlogger "github.com/gobuffalo/mw-paramlogger"
+	"github.com/unrolled/secure"
+
+	"github.com/gobuffalo/buffalo-pop/pop/popmw"
+	contenttype "github.com/gobuffalo/mw-contenttype"
+	"github.com/gobuffalo/x/sessions"
+	"github.com/rs/cors"
+
+	"go.elastic.co/apm/module/apmbuffalo"
+	"go.elastic.co/apm/module/apmbuffalo/example/models"
+)
+
+// ENV is used to help switch settings based on where the
+// application is being run. Default is "development".
+var ENV = envy.Get("GO_ENV", "development")
+var app *buffalo.App
+
+// App is where all routes and middleware for buffalo
+// should be defined. This is the nerve center of your
+// application.
+//
+// Routing, middleware, groups, etc... are declared TOP -> DOWN.
+// This means if you add a middleware to `app` *after* declaring a
+// group, that group will NOT have that new middleware. The same
+// is true of resource declarations as well.
+//
+// It also means that routes are checked in the order they are declared.
+// `ServeFiles` is a CATCH-ALL route, so it should always be
+// placed last in the route declarations, as it will prevent routes
+// declared after it to never be called.
+func App() *buffalo.App {
+	if app == nil {
+		app = buffalo.New(buffalo.Options{
+			Env:          ENV,
+			SessionStore: sessions.Null{},
+			PreWares: []buffalo.PreWare{
+				cors.Default().Handler,
+			},
+			SessionName: "_example_session",
+		})
+
+		// Automatically redirect to SSL
+		app.Use(forceSSL())
+
+		// Log request parameters (filters apply).
+		app.Use(paramlogger.ParameterLogger)
+
+		// Set the request content type to JSON
+		app.Use(contenttype.Set("application/json"))
+
+		// Wraps each request in a transaction.
+		//  c.Value("tx").(*pop.Connection)
+		// Remove to disable this.
+		app.Use(popmw.Transaction(models.DB))
+
+		app.GET("/", apmbuffalo.Wrap(buffalo.WrapBuffaloHandlerFunc(HomeHandler), "/"))
```

Comment:
Is there any reason why tracing cannot be implemented as a `buffalo.MiddlewareFunc`? Ideally, instrumenting the requests should be as simple as:

```go
app.Use(apmbuffalo.Middleware())
```

Additionally, it would probably make sense to have an "Instrument" function, like in apmgorilla, which sets the NotFound/MethodNotAllowed handlers as below. Then it would look like:

```go
apmbuffalo.Instrument(app)
```